library/src/test/java/com/novoda/downloadmanager/FileNameExtractorTest.java

```diff
@@ -0,0 +1,27 @@
+package com.novoda.downloadmanager;
+
+import org.junit.Test;
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class FileNameExtractorTest {
+
+    @Test
+    public void returnsFileName_whenAssetUrlOnlyContainsFileName() {
+        String assetUrl = "10MB.zip";
+
+        FileName fileName = FileNameExtractor.extractFrom(assetUrl);
+
+        assertThat(fileName).isEqualTo(LiteFileName.from("10MB.zip"));
+    }
+
+    @Test
+    public void returnsFileName_whenAssetUrlContainsSubdirectories() {
+        String assetUrl = "foo/bar/10MB.zip";
+
+        FileName fileName = FileNameExtractor.extractFrom(assetUrl);
+
+        assertThat(fileName).isEqualTo(LiteFileName.from("10MB.zip"));
```

Comment:
The content of this test is exactly as the previous one... 
it doesn't show that it contains subdirs

Reply:
`returnsFileName_whenAssetUrlContainsSubdirectories`

That's the point of the test. Given an asset url that contains subdirectories `/` it should ONLY return the filename.

Reply:
Ahhh my bad! I'm so used to the `given_when_then` notation in tests names that I quickly read "should contain subdirs", while it's the opposite!
👍  for the test working
👎  for the test naming

Reply:
Everyone had the opportunity to comment on the test naming a while back and it was decided to go with this one. Personally I prefer this because it isn't nearly as verbose and states exactly what the output is.

Reply:
Oh really? Ok then, not a big deal to me ¯\_(ツ)_/¯